sql/index/cardinality_index.go

```diff
@@ -0,0 +1,18 @@
+package index
+
+import "github.com/dolthub/go-mysql-server/sql"
+
+type cardinalityIndex struct {
+    table string
+    index string
+    stats *sql.IndexStats
+}
+
+func NewCardinalityIndex(table, index string, stats *sql.IndexStats) sql.StatisticsIndex {
```

Comment:
I don't see this function used anywhere either. Can this be removed? Seems like you want to have the existing index implementations Max mentioned in https://github.com/dolthub/dolt/issues/5200 and have them implement `StatisticsIndex` instead of creating this new `cardinalityIndex` type.